task/runner.go

```diff
@@ -1,70 +1,58 @@
 package task
 
 import (
-	"encoding/json"
-	"fmt"
-	"github.com/google/uuid"
-	"github.com/turbot/steampipe/constants"
-	"github.com/turbot/steampipe/db"
-	"io/ioutil"
-	"os"
-	"path/filepath"
+	"sync"
 	"time"
+
+	"github.com/turbot/steampipe/db"
+	"github.com/turbot/steampipe/statefile"
 )
 
 const minimumMinutesBetweenChecks = 1440 // 1 day
-const updateStateFileName = "update-check.json"
 
 type Runner struct {
-	currentState state
+	currentState *statefile.State
 	shouldRun    bool
 }
 
 func NewRunner() *Runner {
 	r := new(Runner)
-	r.loadState()
+	r.currentState, _ = statefile.LoadState()
 	r.shouldRun = r.getShouldRun()
 	return r
 }
 
 func (r *Runner) Run() {
 	if r.shouldRun {
+		waitGroup := sync.WaitGroup{}
+		waitGroup.Add(3)
+
 		// check whether an updated version is available
-		checkVersion(r.currentState.InstallationID)
+		go r.runAsyncJob(func() { checkSteampipeVersion(r.currentState.InstallationID) }, &waitGroup)
+
+		// check whether an updated version is available
+		go r.runAsyncJob(func() { checkPluginVersions(r.currentState.InstallationID) }, &waitGroup)
+
 		// remove log files older than 7 days
-		db.TrimLogs()
+		go r.runAsyncJob(func() { db.TrimLogs() }, &waitGroup)
+
 		// update last check time
-		r.updateState()
+		waitGroup.Wait()
+		r.currentState.Save()
+		r.currentState, _ = statefile.LoadState()
 	}
 }
 
-func (r *Runner) loadState() {
-	stateFilePath := filepath.Join(constants.InternalDir(), updateStateFileName)
-	// get the state file
-	_, err := os.Stat(stateFilePath)
-	if err != nil {
-		r.currentState = r.createState()
-		return
-		// create folder structure if not there
-	}
-
-	stateFileContent, err := ioutil.ReadFile(stateFilePath)
-	if err != nil {
-		fmt.Println("Could not read update state file")
-		r.currentState = r.createState()
-		return
-	}
-
-	err = json.Unmarshal(stateFileContent, &r.currentState)
-	if err != nil {
-		fmt.Println("Could not parse update state file")
-		r.currentState = r.createState()
-		return
-	}
+func (r *Runner) runAsyncJob(job func(), wg *sync.WaitGroup) {
+	job()
+	wg.Done()
 }
 
 func (r *Runner) getShouldRun() bool {
 	now := time.Now()
+	if r.currentState == nil {
```

Comment:
is this right - if there is no current state shouldn't we run for the first time?